File: 1111. Maximum Nesting Depth of Two Valid Parentheses Strings/maxDepthAfterSplit.java
class maxDepthAfterSplit {
    public int[] solution(String seq) {
    	int[] res = new int[seq.length()];
    	for (int i = 1; i < res.length; i++) {
    		if (seq.charAt(i) == seq.charAt(i-1)) {
    			res[i] = 1 - res[i - 1];
    		}
    		else {
    			res[i] = res[i - 1];
    		}
    	}
    	return res;
    }

    public static void main(String[] argv) {
    	String str = "()(())()";
    	maxDepthAfterSplit obj = new maxDepthAfterSplit();
    	int[] res = obj.solution(str);
    	for (int i : res) {
    		System.out.print(i + " ");
    	}
    	System.out.println(" ");
    }
}

File: 1111. Maximum Nesting Depth of Two Valid Parentheses Strings/maxDepthAfterSplit.py
class maxDepthAfterSplit:
    def solution(self, seq: str):
        result = [0] * len(seq)
        for i in range(0, len(result)):
            if seq[i] == seq[i-1]:
                result[i]=1-result[i-1]
            else:
                result[i]=result[i-1]
        return result

obj = maxDepthAfterSplit()
string = "(()())"
res = obj.solution(string)
print(res)